src/lein_monolith/plugin.clj

```diff
@@ -18,71 +40,101 @@
 
 
 (defn- add-profile-paths
-  "Update a profile paths entry by adding the absolute paths from the given
-  project. Returns the updated profile."
+  "Update a profile paths entry by adding the paths from the given project.
+  Returns the updated profile."
   [profile project k]
-  (update profile k into
-          (map (partial str (:root project) "/")
-               (get project k))))
-
-
-(defn merged-profile
-  "Constructs a profile map containing merged (re)source and test paths."
-  [subprojects]
-  (reduce-kv
-    (fn [profile _ project]
-      (-> profile
-          (add-profile-paths project :resource-paths)
-          (add-profile-paths project :source-paths)
-          (add-profile-paths project :test-paths)))
-    {:dependencies (subproject-dependencies subprojects)
-     :resource-paths []
-     :source-paths []
-     :test-paths []}
-    subprojects))
+  (update profile k #(with-meta (distinct (concat % (get project k))) (meta %))))
```

Comment:
Using `into` was causing the meta to meta to get lost for some reason. Explicitly concatting and reattaching the meta.